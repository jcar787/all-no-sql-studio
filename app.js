import express from 'express';
import { MySql } from './lib';
import { router as connectionsRouter } from './connections';
import morgan from 'morgan';

export const app = express();

app.disable('x-powered-by');

// mw
app.use(express.json());
app.use(express.urlencoded({ extended: true }));
app.use(morgan('combined'));
app.use((req, res, next) => {
  console.log(req.body);
  next();
});

// routes
app.use('/connections', connectionsRouter);

app.get('/', async (req, res) => {
  res.send("It's working");
});

app.get('/databases', async (req, res) => {
  try {
    const mysql = new MySql();
    await mysql.connect();
    const databases = await mysql.getDatabases();
    res.send(databases);
  } catch (error) {
    res.send(error.message);
  }
});

app.use((req, res, next) => {
  return res.status(404).end();
});
