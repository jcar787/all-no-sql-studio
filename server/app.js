import express from 'express';
import { MySql } from './lib';
import { router as connectionsRouter } from './connections';
import { router as databasesRouter } from './databases';
import morgan from 'morgan';

export const app = express();

app.disable('x-powered-by');

// mw
app.use(express.json());
app.use(express.urlencoded({ extended: true }));
app.use(morgan('combined'));
app.use((req, res, next) => {
  console.log(req.body);
  next();
});

// routes
app.use('/connections', connectionsRouter);
app.use('/databases', databasesRouter);

app.get('/', async (req, res) => {
  res.send("It's working");
});

app.use((req, res) => {
  return res.status(404).end();
});
