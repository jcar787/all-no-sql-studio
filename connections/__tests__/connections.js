import request from 'supertest';
import { app } from '../../app';

describe('Test connections API', () => {
  describe('GET /connections', () => {
    test('/connections should return a list of connections', async () => {
      const response = await request(app).get('/connections');
      expect(response.statusCode).toBe(200);
      expect(response.body).toMatchSnapshot();
    });

    // describe('GET /connections/:name', () => {
    //   test('/connections/mysql should return one connection');
    //   test('/connections/dontexist should return 404');
    // });
  });

  //   describe('POST /connections', () => {
  //     test('/connections should create a new connection');
  //     test('/connections should throw an error connection exists');
  //     test('/connections/:name should return created connection');
  //   });

  //   describe('PUT /connections/:name', () => {
  //     test('/connections/:name should update a connection');
  //     test('/connections/:name should return 404');
  //   });

  //   describe('DELETE /connections/:name', () => {
  //     test('/connections/:name should delete a connection');
  //     test('/connections/:name should return 404');
  //   });
});
