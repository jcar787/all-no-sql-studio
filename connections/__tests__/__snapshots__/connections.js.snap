// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test connections API GET /connections /connections should return a list of connections 1`] = `
Object {
  "mongo": Object {
    "host": "localhost",
    "password": "",
    "port": 27107,
    "type": "mongo",
    "username": "root",
  },
  "mysql": Object {
    "host": "localhost",
    "password": "",
    "port": 3306,
    "type": "mysql",
    "username": "root",
  },
  "newconn": Object {
    "host": "localhost",
    "password": "",
    "port": 3306,
    "type": "mysql",
    "username": "root",
  },
}
`;

exports[`Test connections API GET /connections GET /connections/:name /connections/mysql should return one connection 1`] = `
Object {
  "host": "localhost",
  "password": "",
  "port": 3306,
  "type": "mysql",
  "username": "root",
}
`;
